src/components/Icon/Icon.tsx

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+
+type IntrinsicElements = JSX.IntrinsicElements["svg"];
+
+interface Props extends IntrinsicElements {
+  name: string;
```

Comment:
Not required, just a thought.

If you wanted to make `iconChoices` a type to be used on the name prop, you could do it like this.

```suggestion
import React from "react";
import { iconChoices } from "./IconChoices";

type IconChoices = typeof iconChoices[number];
type IntrinsicElements = JSX.IntrinsicElements["svg"];

interface Props extends IntrinsicElements {
  name: IconChoices;
```

Reply:
I like it!